quickwit-metastore/src/metastore/single_file_metastore.rs

```diff
@@ -28,16 +28,25 @@ use quickwit_storage::{
     quickwit_storage_uri_resolver, Storage, StorageErrorKind, StorageResolverError,
     StorageUriResolver,
 };
+use serde::{Deserialize, Serialize};
 use tokio::sync::Mutex;
 
 use crate::checkpoint::CheckpointDelta;
 use crate::metastore::match_tags_filter;
 use crate::{
-    IndexMetadata, MetadataSet, Metastore, MetastoreError, MetastoreFactory,
-    MetastoreResolverError, MetastoreResult, SplitMetadata, SplitMetadataAndFooterOffsets,
-    SplitState,
+    IndexMetadata, Metastore, MetastoreError, MetastoreFactory, MetastoreResolverError,
+    MetastoreResult, SplitMetadata, SplitMetadataAndFooterOffsets, SplitState,
 };
 
+/// A MetadataSet carries an index metadata and its split metadata.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MetadataSet {
+    /// Metadata specific to the index.
+    pub index: IndexMetadata,
+    /// List of splits belonging to the index.
+    pub splits: HashMap<String, (SplitMetadataAndFooterOffsets, SplitState)>,
```

Comment:
Not necessarily in this PR, but I think it's a good time to discuss.
Maybe a struct type to wrap the tuple `(SplitMetadataAndFooterOffsets, SplitState)` as we also need to pull out `create_timestamp` & `update_timestamp`

Reply:
I think we might want to pull out update_timestamp. (create_timestamp too if it helps)

Reply:
I'd rather have a struct than a tuple here.